src/vehicle_cmd.cpp

```diff
@@ -135,7 +135,7 @@ CommandCost CmdBuildVehicle(TileIndex tile, DoCommandFlag flags, uint32 p1, uint
 	/* Vehicle construction needs random bits, so we have to save the random
 	 * seeds to prevent desyncs. */
 	SavedRandomSeeds saved_seeds;
-	if (flags != subflags) SaveRandomSeeds(&saved_seeds);
+	SaveRandomSeeds(&saved_seeds);
```

Comment:
Is this really correct? Is it okay to save the seeds even if DC_EXEC was added to the subflags in line 133?

Reply:
It's only restored if `flags != subflags` (or, if DC_EXEC was added)

Reply:
But specifically, GCC isn't clever enough to realise that it's the same condition for RestoreRandomSeeds, which flags up "may be used uninitialised" warnings in that function without it. Making subflags `const` doesn't seem to help